### ISSUE TITLE
Error in tcrossprod(x, y)

### ISSUE THREAD

**Issue by nickmckay** [nickmckay/GeoChronR#79]
Moving this here from lipdR issues - @cknightgov

> Actually runBacon() does create an error, I had missed it previously:
> Error in tcrossprod(x, y) :
> requires numeric/complex matrix/vector arguments
> 
> I tried L = runBacon(L) and L2 = runBacon(L) and got the same error message about the tcrossprod(x, y).

Seems like this is an issue with a specific dataset - is it one that's available publicly? Could you share the code you used to produce this and we'll look into it?

Thanks!

**Comment by cknightgov** [comment]
The error is produced with multiple datasets, both my own and ones from your tutorial. Here is one example that produces the error in runBacon() from your vignette:

K <- lipdR::readLipd("http://lipdverse.org/geoChronR-examples/Kurupa.Boldt.2015.lpd")

K <- runBacon(K,
              lab.id.var = 'labID', 
              age.14c.var = 'age14C', 
              age.14c.uncertainty.var = 'age14CUncertainty',
              age.var = 'age', 
              age.uncertainty.var = 'ageUncertainty', 
              depth.var = 'depth', 
              reservoir.age.14c.var = NULL, 
              reservoir.age.14c.uncertainty.var = NULL, 
              rejected.ages.var = NULL,
              bacon.acc.mean = 10,
              bacon.thick = 7,
              ask = FALSE,
              bacon.dir = "~/Cores",
              suggest = FALSE,
              close.connection = FALSE)

In the meantime, is there a way for me to use the version of geoChron from last week? I had no trouble with using runBacon then, and I would love to keep working on my project. 

Thanks again for your prompt replies!

**Comment by cknightgov** [comment]
Just as an update, I started using R Studio in the posit cloud and I was able to execute runBacon on the Kurupa tutorial core (yay!). I got the data by running: K <- lipdR::readLipd("http://lipdverse.org/geoChronR-examples/Kurupa.Boldt.2015.lpd")

However, if I download a dataset and try to upload to R, I get a new error: 

lipdR::readLipd("~/Downloads/Kurupa.Boldt.2015.lpd")

[1] "Loading 1 datasets from ~/Downloads/Kurupa.Boldt.2015 (1).lpd..."
[1] "reading: Kurupa.Boldt.2015 (1).lpd"
################################################################################################

################################################################################################

### Archive Type ###
### Geographic Metadata ###
Error in if (!is.finite(L$geo$elevation)) { : argument is of length zero

I've tried with a few datasets and the hangup consistently happens at the L$geo$elevation step.

**Comment by nickmckay** [comment]
I wasn't able to replicate the `tcrossprod()` error, which is weird. 

For this second error, it looks like that's actually an issue with the print.lipd() function in lipdR. But you should be able to avoid that by assigning that to a variable when you load it in:

`K <- lipdR::readLipd("~/Downloads/Kurupa.Boldt.2015.lpd")`

**Comment by cknightgov** [comment]
Ok. I ran K <- lipdR::readLipd("~/Downloads/Kurupa.Boldt.2015.lpd"), but there is no data in the file. It's a list of 1.

For example, the next step in the tutorial is the following, which produces this error:
K <- lipdR::readLipd("~/Downloads/Kurupa.Boldt.2015.lpd")
sp <- plotSummary(K, paleo.data.var = "RABD",summary.font.size = 6)
Error in plotSummary(K, paleo.data.var = "RABD", summary.font.size = 6) : 
  This is either not a single LiPD object, or it has no dataSetName. plotSummary requires a single LiPD object as input 

I can see in the R environment that Kurupa data from the lipdverse is a list of 13, but Kurupa data from downloads is a list of 1, even when assigning it to a variable.

**Comment by nickmckay** [comment]
Is there a LiPD file on your computer at
"~/Downloads/Kurupa.Boldt.2015.lpd"? It looks like some of our updates to
lipdR make it try to return a LiPD object even if the file doesn't exist.
I'll put an issue there.

**Comment by cknightgov** [comment]
Yes I verified that I can see the Kurupa lipd file in my downloads. 

<img width="254" alt="Screenshot 2023-09-15 at 12 48 56 PM" src="https://github.com/nickmckay/GeoChronR/assets/144858461/12b3f722-80f3-493d-bc4a-6c8c73941e2d">

**Comment by nickmckay** [comment]
That's strange. Can you load it into your local version of R (not posit R cloud) using:

`K <- lipdR::readLipd("http://lipdverse.org/geoChronR-examples/Kurupa.Boldt.2015.lpd")`

Because that should work locally as well.

**Comment by cknightgov** [comment]
Yes that has consistently worked, and I just re-confirmed on my local version of R.

**Comment by cknightgov** [comment]
I still cannot get runBacon() to work locally, even though it works on posit R cloud. Here is the code with the error message:

> K <- lipdR::readLipd("http://lipdverse.org/geoChronR-examples/Kurupa.Boldt.2015.lpd")
trying URL 'https://lipdverse.org/geoChronR-examples/Kurupa.Boldt.2015.lpd'
Content type 'unknown' length 25533 bytes (24 KB)
==================================================
downloaded 24 KB

[1] "Loading 1 datasets from /var/folders/ly/zxcknvtd3nn987yvq6m_r3zmgvznk_/T//Rtmpj04Wwi/Kurupa.Boldt.2015.lpd..."
[1] "reading: Kurupa.Boldt.2015.lpd"

> K <- runBacon(K,
+               lab.id.var = 'labID', 
+               age.14c.var = 'age14C', 
+               age.14c.uncertainty.var = 'age14CUncertainty',
+               age.var = 'age', 
+               age.uncertainty.var = 'ageUncertainty', 
+               depth.var = 'depth', 
+               reservoir.age.14c.var = NULL, 
+               reservoir.age.14c.uncertainty.var = NULL, 
+               rejected.ages.var = NULL,
+               bacon.acc.mean = 10,
+               bacon.thick = 7,
+               ask = FALSE,
+               bacon.dir = "~/Cores",
+               suggest = FALSE,
+               close.connection = FALSE)
Looking for laboratory ID 
[1] "Found it! Moving on..."
Looking for radiocarbon ages 
[1] "Found it! Moving on..."
Looking for 1-sigma radiocarbon age uncertainty (+/-) 
[1] "Found it! Moving on..."
Looking for calibrated/calendar ages 
[1] "Found it! Moving on..."
Looking for 2-sigma calibrated age uncertainty (+/-) 
[1] "Found it! Moving on..."
Looking for depth or position 
[1] "Found it! Moving on..."
Looking for radiocarbon reservoir age offsets (deltaR) 
radiocarbon reservoir age offsets (deltaR) does not seem to exist, moving on.
Looking for radiocarbon reservoir age offsets (deltaR) uncertainties 
radiocarbon reservoir age offsets (deltaR) uncertainties does not seem to exist, moving on.
Looking for rejected ages 
rejected ages does not seem to exist, moving on.
[1] "Variable choices for reuse..."
For future reference: here are the options you chose:
 Find later with getLastVarString()
lab.id.var = 'labID', age.14c.var = 'age14C', age.14c.uncertainty.var = 'age14CUncertainty', age.var = 'age', age.uncertainty.var = 'ageUncertainty', depth.var = 'depth', reservoir.age.14c.var = NULL, reservoir.age.14c.uncertainty.var = NULL, rejected.ages.var = NULL, 
  id  age error depth cc dR dSTD ta tb
1  8  -60     1   0.0  0  0    0 33 34
2  7  -13     5   5.7  0  0    0 33 34
3  4  440    20  76.5  1  0    0  3  4
4  5 1470    15 167.5  1  0    0  3  4
5  6 2035    15 224.5  1  0    0  3  4
6  2 3315    20 398.0  1  0    0  3  4
7  3 4020    20 505.5  1  0    0  3  4
8  1 4665    20 593.5  1  0    0  3  4
The run's files will be put in this folder: /Users/cknight/Cores/Kurupa.Boldt.2015
 Using a mix of cal BP and calibrated C-14 dates

Warning! The file with the dates seems newer than the run you are loading. If any dates have been added/changed/removed?, then please run Bacon.cleanup()
 Run Kurupa.Boldt.2015 with 86 sections? (Y/n) 
y
 Will run 16,500,000 iterations and store 4,000
Reading /Users/cknight/Cores/Kurupa.Boldt.2015/Kurupa.Boldt.2015_86.bacon
Constant calibration curve.
IntCal20: Reading from file: /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rintcal/extdata/3Col_intcal20.14C
Marine20: Reading from file: /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rintcal/extdata/3Col_marine20.14C
SHCal20: Reading from file: /Library/Frameworks/R.framework/Versions/4.1/Resources/library/rintcal/extdata/3Col_shcal20.14C
Added det: 8:  -60.0+-1.0    d=0 ResCorr=   0.0+-0.0    a=33 b=34 cc=Constant c. curve
Added det: 7:  -13.0+-5.0    d=5.7 ResCorr=   0.0+-0.0    a=33 b=34 cc=Constant c. curve
Added det: 4:  440.0+-20.0   d=76.5 ResCorr=   0.0+-0.0    a=3 b=4 cc=IntCal20
Added det: 5: 1470.0+-15.0   d=167.5 ResCorr=   0.0+-0.0    a=3 b=4 cc=IntCal20
Added det: 6: 2035.0+-15.0   d=224.5 ResCorr=   0.0+-0.0    a=3 b=4 cc=IntCal20
Added det: 2: 3315.0+-20.0   d=398 ResCorr=   0.0+-0.0    a=3 b=4 cc=IntCal20
Added det: 3: 4020.0+-20.0   d=505.5 ResCorr=   0.0+-0.0    a=3 b=4 cc=IntCal20
Added det: 1: 4665.0+-20.0   d=593.5 ResCorr=   0.0+-0.0    a=3 b=4 cc=IntCal20
BaconFixed: Bacon jumps model with fixed c's.
            K=86, H=0, dim=88, Seed=871097, Dc=7.000000, c(0)=0.000000, c(K)=602.000000

twalk:     16500000 iterations to run, Sun Sep 17 15:01:10 2023
       
BUFSIZ is 1024, optimal block size changed to 4096
twalk: All stored iterations to be saved in file /Users/cknight/Cores/Kurupa.Boldt.2015/Kurupa.Boldt.2015_86.out
twalk: 13407232 iterations so far
Will finish in approx. 7 seconds.
twalk: Finished,  0.3% of moved pars per iteration (ratio 47480.363636/16500000). Output in file /Users/cknight/Cores/Kurupa.Boldt.2015/Kurupa.Boldt.2015_86.out,
      Sun Sep 17 15:01:48 2023

bacon: 1037 WarnBeyondLimits warnings:
bacon: WARNING: calibration attempted beyond MinYr= -73.000000 or MaxYr= 1000000.000000
bacon: burn in (initial iterations which will be removed): 6600000
Eso es to...eso es to...eso es to...eso es toooodo amigos!
Warning, this will take quite some time to calculate. I suggest increasing d.by to, e.g.10

Calculating age ranges...

  |===========================================================================================| 100%
Preparing ghost graph... 
  |===========================================================================================| 100%
Mean 95% confidence ranges 319 yr, min. 4 yr at 0 cm, max. 506 yr between 294 and 308 cm
100% of the dates overlap with the age-depth model (95% ranges)


[1] "taking a short break..."

Error in tcrossprod(x, y) : 
  requires numeric/complex matrix/vector arguments

**Comment by JanPetrik1** [comment]
Hello guys,
I have the same issue. Is there any progress?

**Comment by cknightgov** [comment]
Hello! I still get that error when I use R Studio on my desktop. My work around has been to use Posit Cloud services (it's free) and I have not gotten the error there despite using identical code.
